app/models/response/HateoasResponseBox.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.response
+
+import controllers.routes
+import models.Box
+import play.api.libs.json.{JsNull, JsObject, Json}
+import utils.CallOps._
+
+object HateoasResponseBox {
+
+  def apply(box: Box): JsObject = {
+
+    val boxUrl = routes.PushPullNotificationController.getBoxInfo().urlWithContext
+
+    JsObject(
+      Json
+        .obj(
+          "boxId"                      -> box.boxId,
+          "boxName"                 -> box.boxName,
+          "_links" -> Json.obj(
+            "self"     -> Json.obj("href" -> boxUrl),
+          )
+        )
+        .fields
+        .filter(
+          t => t._2 != JsNull
```

Comment:
Can any of these things be null?

Reply:
No, its just a leftover, I'll clean it out.